lib/service/service.go

```diff
@@ -4542,6 +4542,7 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 	}
 
 	tlscfg := serverTLSConfig.Clone()
+	tlscfg.NextProtos = []string{string(alpncommon.ProtocolHTTP2), string(alpncommon.ProtocolProxySSHGRPC)}
```

Comment:
Are we deliberately preferring `h2` to the custom Teleport next proto?

Reply:
Yeah this is what I was mentioning here https://github.com/gravitational/teleport/pull/43473#issuecomment-2192031355, IIRC the server has to actually select `h2` for the `google.golang.org/grpc` code to work